Client/Frontend/Browser/BrowserViewController/OpenSearch/BrowserViewController+OpenSearch.swift

```diff
@@ -190,13 +190,20 @@ extension BrowserViewController {
     }
     
     private func createSearchEngine(_ url: URL, reference: String, icon: UIImage) {
-        NetworkManager().downloadResource(with: url).uponQueue(.main) { [weak self] response in
-            guard let openSearchEngine = OpenSearchParser(pluginMode: true).parse(
-                    response.data, referenceURL: reference, image: icon, isCustomEngine: true) else {
-                return
+        Task.detached(priority: .userInitiated) {
+            do {
+                let response = try await NetworkManager().downloadResource(with: url)
+                DispatchQueue.main.async { [weak self] in
```

Comment:
Don't use dispatch queues inside Task's, run it on MainActor via `MainActor.run { ... }` or a child task: `Task { @MainActor in ... }`